---
title:    "C++: 표준 오류에 쓰는 것"
keywords: ["C++"]
---

{{< edit_this_page >}}

## 왜

프로그래밍에서 `std::cerr`를 사용해 에러를 출력하는 것의 이유는, 디버깅에 매우 유용하기 때문입니다. 프로그램에서 발생한 오류를 표시하는 것으로, 오류를 찾는 데 도움이 됩니다.

## 사용방법

에러를 `std::cerr`을 사용해 출력하는 것은 매우 간단합니다. 아래의 예시 코드를 참고하세요.

```C++
#include <iostream>
using namespace std;

int main() {
    int a = 20;
    int b = 0;
    int c;

    if (b == 0) {
        // 에러 메시지 출력
        cerr << "b는 0이 될 수 없습니다." << endl;
        return 1;
    }
    c = a / b;
    // c의 값 출력
    cout << "결과: " << c << endl;

    return 0;
}
```

위의 코드에서 `cerr`을 사용한 부분은 오류를 출력하는 부분입니다. 만약 `b`가 0이 되는 경우, 에러 메시지를 출력하고 프로그램을 종료합니다. 결과는 다음과 같게 출력됩니다.

`b는 0이 될 수 없습니다.`

위의 예시 코드를 참고하여 `std::cerr`을 사용해 에러를 출력하는 방법을 익혀보세요.

## 자세히 알아보기

`std::cerr`은 `std::ostream`의 객체입니다. 따라서 `std::cout`과 같이 사용할 수 있습니다. 하지만 `std::cout`과는 달리 `std::cerr`은 버퍼링되지 않기 때문에 바로 출력됩니다. 또한 `std::cerr`은 일반적으로 화면이 아닌 터미널과 같은 장치에 출력되기 때문에, 결과를 확인하기 좋습니다.

`std::cerr`을 사용할 때 주의할 점은, 에러 메시지를 출력할 때는 `endl`을 사용해줘야 합니다. `endl`을 사용함으로써 라인을 종료하여 출력 버퍼를 비워줍니다. 이를 해주지 않으면 에러 메시지가 나타나지 않을 수 있습니다.

## 관련 자료

- [C++ 에러 메시지 출력 방법](http://www.cplusplus.com/reference/ios/cerr)
- [C++ 입출력 스트림](https://dojang.io/mod/page/view.php?id=1109)